acceptance/indy_acceptance/indy-tests/libraries/utils.py

```diff
@@ -0,0 +1,131 @@
+    else:
+        print("Warning: Length of prefix and suffix more than %s chars" % str(size))
+    result = str(prefix) + random_str + str(suffix)
+    return result
+
+
+def exit_if_exception(code):
+    """
+    If "code" is an exception then raise the "code".
+    Unless "code" is an exception then return the "code".
+    :param code: (optional) code that you want to check.
+    :return: "code" if it is not an exception.
+    """
+    if (isinstance(code, IndyError) or (isinstance(code, Exception))):
+        exit(1)
+    else:
+        return code
+
+
+async def perform(steps, func, *agrs):
+    """
+    Execute an function and set status, message for the last test step depend on the result of the function.
+
+    :param steps: (optional) list of test steps.
+    :param func: (optional) executed function.
+    :param agrs: argument of function.
+    :return: the result of function of the exception that the function raise.
+    """
+    try:
+        result = await func(*agrs)
+        steps.get_last_step().set_status(Status.PASSED)
+    except IndyError as E:
+        print(Colors.FAIL + Message.INDY_ERROR.format(str(E)) + Colors.ENDC)
+        steps.get_last_step().set_message(str(E))
+        steps.get_last_step().set_status(Status.FAILED)
+        return E
+    except Exception as Ex:
+        print(Colors.FAIL + Message.EXCEPTION.format(str(Ex)) + Colors.ENDC)
+        steps.get_last_step().set_message(str(Ex))
+        steps.get_last_step().set_status(Status.FAILED)
+        return Ex
+    return result
+
+
+async def perform_with_expected_code(steps, func, *agrs, expected_code=0):
+    """
+    Execute the "func" with expectation that the "func" raise an IndyError that IndyError.error_code = "expected_code".
+
+    :param steps: (optional) list of test steps.
+    :param func: (optional) executed function.
+    :param agrs: arguments of "func".
+    :param expected_code: the error code that you expect in IndyError.
+    :return: exception if the "func" raise it without "expected_code".
+             'None' if the "func" run without any exception of the exception contain "expected_code".
+    """
+    try:
+        await func(*agrs)
+        steps.get_last_step().set_message("Can execute without exception.")
+        steps.get_last_step().set_status(Status.FAILED)
+        return None
+    except IndyError as E:
+        if E.error_code == expected_code:
+            steps.get_last_step().set_status(Status.PASSED)
+            return None
+        else:
+            print(Colors.FAIL + Message.INDY_ERROR.format(str(E)) + Colors.ENDC)
+            steps.get_last_step().set_message(str(E))
+            return E
+    except Exception as Ex:
+        print(Colors.FAIL + Message.EXCEPTION.format(str(Ex)) + Colors.ENDC)
+        return Ex
+
+
+def run_async_method(method):
+    """
+    Run async method until it complete.
+
+    :param method: (optional).
+    """
+    import asyncio
+    loop = asyncio.new_event_loop()
```

Comment:
What is the reason to use `new_event_loop` rather than `get_event_loop` here?
As I can see, we should use `get_event_loop` and later call `close` on the returned event loop only one time - at the upper level in `TestScenarioBase.execute_scenario`.
So it seems that the method `run_async_method` is redundant. We can get an event loop, run each async method on it until complete and then close the event loop in `TestScenarioBase.execute_scenario`.

Reply:
Updated. Please refer to the latest code.